### PR TITLE
Module for managing F5 GTM wide ip

### PR DESCRIPTION
**Pull request by perzizzle**
Ability to set load balancing method of F5 GTM wide ip

### CODE REVIEW

**Comment by gregdek** [comment]
@perzizzle Thanks for submitting this new module.

To help facilitate reviews, we’ve broadened the number of people who can approve modules for inclusion into the Extras repository. The list of official reviewers can be found here:

https://github.com/ansible/ansible-modules-extras/blob/devel/REVIEWERS.md 

Our new policy is that if a new module is reviewed and approved by at least two official module reviewers, and receives no -1 votes, the module will be approved for inclusion.  We will be asking the community of reviewers to take a look at these modules on a regular basis.

To ensure that your module has the best chance of being approved, please double-check that you adhere to the Ansible module guidelines: http://docs.ansible.com/developing_modules.html#module-checklist

**Comment by perzizzle** [comment]
Playbook to validate functionality https://gist.github.com/perzizzle/36da14cbdc607e3eec13 depends on an extra_vars.yml similar to https://gist.github.com/perzizzle/190195da598a0311ce6b run with ansible-playbook main_wide_ip.yml --extra-vars=@extra_vars.yml -vv

**Comment by gregdek** [comment]
@caphrim007 Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

File: network/f5/bigip_gtm_wide_ip.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                result = {'changed': True}
+
+    except Exception, e:
+        module.fail_json(msg="received exception: %s" % e)
+
+    module.exit_json(**result)
+
+# import module snippets
+from ansible.module_utils.basic import *
+main()
```
Comment (caphrim007) [inline network/f5/bigip_gtm_wide_ip.py]: If we could wrap this in an if conditional, that'd be great.

Comment (perzizzle) [inline network/f5/bigip_gtm_wide_ip.py]: Done

File: network/f5/bigip_gtm_wide_ip.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+EXAMPLES = '''
+  - name: Set lb method
+    local_action: >
+      bigip_gtm_wide_ip
+      server=192.168.0.1
+      user=admin
+      password=mysecret
+      lb_method=round_robin
+      wide_ip=my_wide_ip
```
Comment (caphrim007) [inline network/f5/bigip_gtm_wide_ip.py]: In version >= 11.6.0 this needs to be a fully qualified domain name without invalid characters (of which underscore is an invalid char). Can we change this to be my-wide-ip.domain.com or something FQDN-y?

**Comment by gregdek** [comment]
First, adding boilerplate for the new process:

# 

Thanks for submitting this new module to Ansible Extras! This module is now in community review, a process that is open to all Ansible users. In order for this module to be approved, it must gain the following votes:

“works_for_me”: If you have tested the module thoroughly, including testing of all of the module’s options, and if the module works for you, please add “works_for_me” in the comments.

“passes_guidelines”: If you have gone through the module guidelines and the module meets all of the requirements, please add “passes_guidelines” in the comments. Guidelines are available here: http://docs.ansible.com/developing_modules.html#module-checklist

“needs_revision”: If the module fails to work for you, or if it doesn’t meet guidelines, please add “needs_revision” in the comments with details about what needs to be fixed.

When a module has both “works_for_me” and “passes_guidelines” tags, we will promote the module for inclusion in Ansible Extras.  At this point, you will be expected to maintain the module by fixing bugs and evaluating pull requests in a timely manner. 

Thanks again for submitting your Ansible module!

**Comment by caphrim007** [comment]
works_for_me passes_guidelines

**Comment by gregdek** [comment]
There we go. Thanks @caphrim007 for the reviews, and thanks @perzizzle for the new module!

Marking for inclusion. cc: @ansible/core team.

**Pull request merged**